web/src/components/troubleshoot/SupportBundleRow.jsx

```diff
@@ -93,20 +93,26 @@ class SupportBundleRow extends React.Component {
                 </div>
                 <div className="flex u-marginTop--10">
                   {bundle?.analysis?.insights?.length ?
-                    <div className="flex flex1 u-marginRight--5">
-                      {sortBy(filter(bundle?.analysis?.insights, (i) => i.level !== "debug"), ["desiredPosition"]).map((insight, i) => (
-                        <div key={i} className="analysis-icon-wrapper">
-                          {insight.icon_key ?
-                            <span className={`icon clickable analysis-${insight.icon_key}`} data-tip={`${bundle.id}-${i}-${insight.key}`} data-for={`${bundle.id}-${i}-${insight.key}`}></span>
-                            : insight.icon ?
-                              <span className="u-cursor--pointer" style={{ backgroundImage: `url(${insight.icon})` }} data-tip={`${bundle.id}-${i}-${insight.key}`} data-for={`${bundle.id}-${i}-${insight.key}`}></span>
-                              : null
-                          }
-                          <ReactTooltip id={`${bundle.id}-${i}-${insight.key}`} effect="solid" className="replicated-tooltip">
-                            <span>{insight.detail}</span>
-                          </ReactTooltip>
-                        </div>
-                      ))}
+                    <div className="flex flex1 u-marginRight--5 alignItems--center">
+                      {sortBy(filter(bundle?.analysis?.insights, (i) => i.level !== "debug"), ["desiredPosition"]).map((insight, i) => {
+                        let iconObj;
+                        if (insight.icon) {
+                          iconObj = parseIconUri(insight.icon);
```

Comment:
I know this is a small change but I will suggest creating a function for this. And call that function in a render.
Right now, there are too many if statements. 
```

renerInsightIcon = (insight) => {
  if(insight.icon) {
    return ....
   } else {
    return ...
  }
}
```

Reply:
@jgruica i'm not sure I follow. All that's doing is running a utility if it has an iconURI. Seems like a very verbose change to decided wether or not to run that utility

Reply:
Yeah I got that. I'm just saying that maybe it will be more readable if all of this is in function.
For example. 
On line 99 you're adding if (insight.icon) and in line 104. you're doing again if statement which is checking the same thing.

`  renderInsightIcon = (bundle, i, insight) => {
    if (insight.icon) {
      let iconObj = parseIconUri(insight.icon);
      return (
        <div className="tile-icon" style={{ backgroundImage: `url(${iconObj.uri})`, width: `${iconObj.dimensions?.w}px`, height: `${iconObj.dimensions?.h}px` }} data-tip={`${bundle.id}-${i}-${insight.key}`} data-for={`${bundle.id}-${i}-${insight.key}`}></div>
      )
    } else {
      return (
        <span className={`icon clickable analysis-${insight.icon_key}`} data-tip={`${bundle.id}-${i}-${insight.key}`} data-for={`${bundle.id}-${i}-${insight.key}`}></span>
      )
    }
  }`

with function you can just call `{this.renderInsightIcon(bundle, i, insight)}` in line 104.

Reply:
I see, yeah I can update that